server/services/v1/api.go

```diff
@@ -122,13 +124,14 @@ func (s *apiService) BeginTransaction(ctx context.Context, r *api.BeginTransacti
 		return nil, err
 	}
 
-	_, txCtx, err := s.txMgr.StartTx(ctx, true)
+	session, err := s.querySessExecutor.CreateQuerySession(ctx, true)
 	if err != nil {
 		return nil, err
 	}
+	s.querySessExecutor.AddQuerySession(session.txCtx.Id, session)
```

Comment:
Can this be automatically called by Create?

Reply:
We can do that, wanted to avoid an extra param to be passed in the function but I think it is fine.